File: server/routes/vehicle.js

const auth = require('../middlewares/auth');
const Vehicle = require('../models/vehilce')
const express = require('express');
const vehicleRouter = express.Router();


vehicleRouter.post("/api/addVehicle", async (req, res) => {
    try {
        const { name, description, price, images, brand } = req.body;
        let vehicle = Vehicle({
            name, description, price, brand, images
        });
        const vehicleAddRes = await vehicle.save();
        console.log(vehicle);
        return res.json({ vehicle });
    } catch (error) {
        return res.status(500).json({ error: error.message });
    }

});

vehicleRouter.get("/api/getVehicle",auth, async (req, res) => {
    try {
       

        const vehicles = await Vehicle.find();
        if (!vehicles) return res.json([]);
        return res.json({ vehicles });
    } catch (error) {
        return res.status(500).json({ error: error.message });
    }
})



module.exports = vehicleRouter;

File: lib/model/vehicle.dart

import 'package:get/get.dart';

class VehicleModel {
    RxList<Vehicle> data;

    VehicleModel({
        required this.data,
    });

    factory VehicleModel.fromJson(Map<String, dynamic> json) => VehicleModel(
        data: RxList<Vehicle>.from(json["vehicles"].map((x) => Vehicle.fromJson(x))),
    );

    Map<String, dynamic> toJson() => {
        "vehicles": List<dynamic>.from(data.map((x) => x.toJson())),
    };
}

class Vehicle {
    String name;
    String images;
    String brand;
    String description;
    String price;

    Vehicle({
        required this.name,
        required this.images,
        required this.brand,
        required this.description,
        required this.price,
    });

    factory Vehicle.fromJson(Map<String, dynamic> json) => Vehicle(
        name: json["name"],
        images: json["images"],
        brand: json["brand"],
        description: json["description"],
        price: json["price"],
    );

    Map<String, dynamic> toJson() => {
        "name": name,
        "images": images,
        "brand": brand,
        "description": description,
        "price": price,
    };
}
